src/rgw/setup/rgw_service.py

```diff
@@ -31,14 +31,20 @@ def start(conf: MappedConf, config_file, log_file, motr_trace_dir, index: str =
         try:
             os.environ['M0_TRACE_DIR'] = motr_trace_dir
             cmd = "/usr/bin/radosgw"
-            args = f"-f --name client.rgw-{index} -c {config_file} --no-mon-config &> {log_file}"
-            args = shlex.split(args)
+            args = f" -f --name client.rgw-{index} -c {config_file} --no-mon-config 2>1"
+            time_stamp = """ | gawk '{ print strftime("[%Y-%m-%d %H:%M:%S]"), $0 }' """
+            redirect = f" | tee {log_file} "
+            final_cmd = cmd + args + time_stamp + redirect
             sys.stdout.flush()
             sys.stderr.flush()
-            os.execl(cmd, cmd, *args)  # nosec
+            with open('command_file.sh','w') as fd:
+                fd.write(final_cmd)
+            os.execl('/bin/bash', '/bin/bash', 'command_file.sh')
```

Comment:
so this python process would be overridden by a bash process now and ps would should command_file.sh ! rename it to radosgw_start.sh atleast.

Ideally we want to keep RGW process entrypoint similar to what any ceph rgw user would use it like in Ceph. But you are saying os.execl() doesnt allow piping of commands like above ?

@ujjwalpl fyi

Reply:
@cdeshmukh , changes the filename to `radosgw_start.sh` and the command exits with the same exit status of the last failed command in the pipe. {ideally exit status of `/usr/bin/radosgw` until `gawk` or `tee` errors out}

Reply:
@cdeshmukh I am not recalling this exactly but in general parent process should be in wait state till its child process terminates. Once child process ( in our case its rgw service script) terminates then parent process will also get terminated and kubernetes will try to start it entry point script again.
In case of child process failure, with help of os.execl() command we were able to transfer the control from child process to parent process.

@rohit-k-dwivedi could you please confirm if this behaviour still same with this change.
1) run ps -ef output inside rgw container and check process trees with pid values.
2) kill the rgw service pid and check if container gets terminated or not.

Reply:
process started by rgw miniprov start phase `/bin/bash command_file.sh` takes the pid 1 as expected, 
and killing the process `/usr/bin/radosgw -f --name client.rgw-1 -c /etc/cortx/rgw/bdf3c728d79d4047a80bddfde1b7078c/cortx_rgw.conf --no-mon-config` restarts the pod
```
[root@cortx-server-headless-svc-ssc-vm-rhev4-2782 /]# ps -ef
UID         PID   PPID  C STIME TTY          TIME CMD
root          1      0  0 May30 ?        00:00:00 /bin/bash command_file.sh
root         10      1  0 May30 ?        00:00:44 /usr/bin/radosgw -f --name client.rgw-1 -c /etc/cortx/rgw/bdf3c728d79d4047a80bddfde1b7078c/cortx_rgw.conf --no-mon-config
root         11      1  0 May30 ?        00:00:00 gawk { print strftime("[%Y-%m-%d %H:%M:%S]"), $0 }
root         12      1  0 May30 ?        00:00:00 /usr/bin/coreutils --coreutils-prog-shebang=tee /usr/bin/tee /etc/cortx/log/rgw/bdf3c728d79d4047a80bddfde1b7078c/rgw_startup.log
root       1306      0  0 03:36 ?        00:00:00 /usr/sbin/crond start
root       1769      0  1 04:22 pts/0    00:00:00 /bin/bash
root       1787   1769  0 04:22 pts/0    00:00:00 ps -ef
[root@cortx-server-headless-svc-ssc-vm-rhev4-2782 /]# killall -9 /usr/bin/radosgw
[root@cortx-server-headless-svc-ssc-vm-rhev4-2782 /]# command terminated with exit code 137
[root@ssc-vm-rhev4-2782 cortx-rgw-integration]# kubectl exec -it cortx-server-ssc-vm-rhev4-2782-7c46b5c4bd-cmwbt -c cortx-rgw -- /bin/bash
[root@cortx-server-headless-svc-ssc-vm-rhev4-2782 /]# exit
[root@ssc-vm-rhev4-2782 cortx-rgw-integration]# kubectl get pods
NAME                                              READY   STATUS    RESTARTS        AGE
cortx-consul-client-zcsl8                         1/1     Running   0               14h
cortx-consul-server-0                             1/1     Running   0               14h
cortx-control-864bf5d4-c9gwc                      1/1     Running   0               14h
cortx-data-ssc-vm-rhev4-2782-df8dc48f9-99ng4      4/4     Running   0               14h
cortx-ha-675f585c99-6gnnl                         3/3     Running   0               14h
cortx-kafka-0                                     1/1     Running   1 (14h ago)     14h
cortx-server-ssc-vm-rhev4-2782-7c46b5c4bd-cmwbt   2/2     Running   1 (3m22s ago)   14h
cortx-zookeeper-0                                 1/1     Running   0               14h

```